flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClientFactory.java

```
@@ -81,6 +91,57 @@ private YarnClusterDescriptor getClusterDescriptor(Configuration configuration)
+                            long o1Id = o1.getApplicationId().getId();
+                            long o2Id = o2.getApplicationId().getId();
+                            if (o1Time > o2Time) {
+                                return -1;
+                            } else if (o1Time == o2Time) {
+                                return (o1Id < o2Id) ? 1 : -1;
+                            } else {
+                                return 1;
+                            }
+                        });
+                if (CollectionUtils.isNotEmpty(applicationReports)) {
+                    for (ApplicationReport applicationReport : applicationReports) {
+                        if (StringUtils.equals(
+                                applicationReport.getName(),
+                                configuration.get(YarnConfigOptions.APPLICATION_NAME))) {
+                            if (configuration.contains(YarnConfigOptions.APPLICATION_QUEUE)) {
+                                if (StringUtils.equals(
+                                        applicationReport.getQueue(),
+                                        configuration.get(YarnConfigOptions.APPLICATION_QUEUE))) {
+                                    configuration.setString(
+                                            YarnConfigOptions.APPLICATION_ID,
+                                            applicationReport.getApplicationId().toString());
+                                    break;
+                                }
+                            } else {
+                                configuration.setString(
+                                        YarnConfigOptions.APPLICATION_ID,
+                                        applicationReport.getApplicationId().toString());
+                                break;
+                            }
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
```

In case of a failure, we should propagate the exception.